components/forms/Form/Form.tsx

```diff
@@ -19,15 +25,50 @@ declare global {
  * @param props
  */
 const InnerForm = (props: InnerFormProps & FormikProps<FormValues> & DynamicFormProps) => {
-  const { children, handleSubmit, t, isSubmitting } = props;
+  const { children, handleSubmit, t, isSubmitting, isReCaptchaEnableOnSite } = props;
   const [canFocusOnError, setCanFocusOnError] = useState(false);
   const [lastSubmitCount, setLastSubmitCount] = useState(0);
-
   const errorList = props.errors ? getErrorList(props) : null;
   const errorId = "gc-form-errors";
   const serverErrorId = `${errorId}-server`;
   const formStatusError = props.status === "Error" ? t("server-error") : null;
 
+  const handleFormSubmission = (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    window.grecaptcha.ready(() => {
+      // get reCAPTCHA response
+      window.grecaptcha
+        .execute(process.env.NEXT_PUBLIC_RECAPTCHA_V3_SITE_KEY, { action: "submit" })
+        .then(async (clientToken: string) => {
```

Comment:
**Code style:** Are we using async functions (`await`), or promises (`.then()`).

I think in most other code, we've been moving to using `await`.

@omartehsin1?